packages/frontend/src/os.ts

```diff
@@ -82,11 +87,13 @@ export const apiWithDialog = (<E extends keyof Misskey.Endpoints = keyof Misskey
 	});
 
 	return promise;
-}) as typeof misskeyApi;
+});
+
+type Unpromise<T> = T extends Promise<infer U> ? U : never;
 
 export function promiseDialog<T extends Promise<any>>(
 	promise: T,
-	onSuccess?: ((res: any) => void) | null,
+	onSuccess?: ((res: Unpromise<T>) => void) | null,
```

Comment:
```suggestion
	onSuccess?: ((res: Awaited<T>) => void) | null,
```

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#the-awaited-type-and-promise-improvements

のほうがいいかもと思ったけどthenだからUnpromiseのほうが都合良いのか？

Reply:
~~特に考えずやったので~~これでもいい気がする